packages/contracts/test/token/erc20/governance-wrapped-erc20.ts

```diff
@@ -0,0 +1,335 @@
+import {expect} from 'chai';
+import {ethers} from 'hardhat';
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
+
+import {
+  TestERC20,
+  TestERC20__factory,
+  GovernanceWrappedERC20,
+  GovernanceWrappedERC20__factory,
+} from '../../../typechain';
+import {OZ_ERRORS} from '../../test-utils/error';
+import {getInterfaceID} from '../../test-utils/interfaces';
+
+export type AccountBalance = {account: string; amount: number};
+
+export type MintSettings = {
+  receivers: string[];
+  amounts: number[];
+};
+
+const existingErc20Name = 'Token';
+const existingErc20Symbol = 'TOK';
+
+const governanceWrappedERC20Name = 'GovernanceWrappedToken';
+const governanceWrappedERC20Symbol = 'gwTOK';
+
+describe('GovernanceWrappedERC20', function () {
+  let signers: SignerWithAddress[];
+  let governanceToken: GovernanceWrappedERC20;
+  let erc20: TestERC20;
+  let TestERC20: TestERC20__factory;
+  let GovernanceWrappedERC20: GovernanceWrappedERC20__factory;
+  let defaultBalances: AccountBalance[];
+
+  let defaultExistingERC20InitData: [string, string, number];
+  let defaultGovernanceWrappedERC20InitData: [string, string, string];
+
+  before(async () => {
+    signers = await ethers.getSigners();
+
+    TestERC20 = await ethers.getContractFactory('TestERC20');
+    GovernanceWrappedERC20 = await ethers.getContractFactory(
+      'GovernanceWrappedERC20'
+    );
+    defaultBalances = [
+      {account: signers[0].address, amount: 123},
+      {account: signers[1].address, amount: 456},
+      {account: signers[2].address, amount: 789},
+    ];
+  });
+
+  beforeEach(async function () {
+    defaultExistingERC20InitData = [existingErc20Name, existingErc20Symbol, 0];
+    erc20 = await TestERC20.deploy(...defaultExistingERC20InitData);
+
+    const promises = defaultBalances.map(balance =>
+      erc20.setBalance(balance.account, balance.amount)
+    );
+    await Promise.all(promises);
+
+    defaultGovernanceWrappedERC20InitData = [
+      erc20.address,
+      governanceWrappedERC20Name,
+      governanceWrappedERC20Symbol,
+    ];
+  });
+
+  describe('initialize:', async () => {
+    it('reverts if trying to re-initialize', async () => {
+      erc20 = await TestERC20.deploy(...defaultExistingERC20InitData);
```

Comment:
put these beforeEach of `describe` blocks.

Reply:
Done here 34b4588e60c6cd7ee5b274eadaa59752939a676e